src/api/app/controllers/source_controller.rb

```diff
@@ -521,7 +521,9 @@ def check_and_remove_repositories!(repositories, full_remove, force = false)
     if !force && error[:error]
       raise RepoDependency, error[:error]
     else
-      error = Project.remove_repositories(repositories, full_remove)
+      opts = {no_write_to_backend: true}
+      opts[:recursive_remove] = true if full_remove
+      error = Project.remove_repositories(repositories, opts)
```

Comment:
Maybe result would be more speaking than error? I assume this won't always return an error;-)